### ISSUE TITLE
The modem works fine offline, but not when looping back through the sound card.

### ISSUE THREAD

**Issue by LieBtrau** [aicodix/modem#2]
Hi,
I decided to give your modem a try, so I downloaded the master code of the different repositories today and compiled the code.
Here is what I did:
```
$ dd if=/dev/urandom of=uncoded.dat bs=1 count=5380
5380+0 records in
5380+0 records out
5380 bytes (5.4 kB, 5.3 KiB) copied, 0.05059 s, 106 kB/s
$ ./modem-master/encode encoded.wav 8000 16 1 uncoded.dat 1800 NOCALL
real PAPR: 7.04375 .. 10.9388 dB
$ ./modem-master/decode decoded.dat encoded.wav 
symbol pos: 2272
coarse cfo: 1800 Hz 
oper mode: 6
call sign:    NOCALL
demod .................................................. done
coarse sfo: -0.00905869 ppm
finer cfo: 1800 Hz 
init Es/N0: 32.1313 dB
$ diff -s uncoded.dat decoded.dat 
Files uncoded.dat and decoded.dat are identical
```
So far the good news.  
Then I connect the headphone output of my sound card to the microphone input.
Afterwards, I play back this file through my sound card, and record it.  
The amplitude of the recorded file is lower, but the spectrogram (checked with Audacity) looks very similar.
```
arecord -c 1 -f S16_LE -r 8000 -d 20 recorded.wav
...
$ aplay encoded.wav 
Playing WAVE 'encoded.wav' : Signed 16 bit Little Endian, Rate 8000 Hz, Mono
...
$ ./modem-master/decode decoded.dat recorded.wav 
symbol pos: 2272
coarse cfo: 1799.94 Hz 
oper mode: 6
call sign:    NOCALL
demod .................................................. done
coarse sfo: 25.0049 ppm
finer cfo: 1799.94 Hz 
init Es/N0: 15.0381 dB
payload decoding error.
...
You can find the audio samples here : [samples.zip](https://github.com/aicodix/modem/files/10367203/samples.zip).

I think it might be interesting to you to know what is going wrong here, because I have no clue.
BTW: I changed the offset to 1800, because that fits better with the 300-3.3kHz audio bandwidth of my HTs that I want to use with this modem in the future.

**Comment by xdsopl** [comment]
There might be something nonlinear happening on the higher end of the frequency, as your recording doesn't show a completely flat spectrum in Audacity.
Can you try higher sample rates and see if that helps?
Lowering the carrier frequency might also help.
Last but not least: try narrower modes, like mode 7 or 11.

**Comment by LieBtrau** [comment]
Hi, thanks for the info.  There was a problem with my test setup, probably some bad connection.

All modes I tested work fine with the audio loopback: [test report](https://github.com/LieBtrau/digital-walkie-talkie/blob/master/doc/PMR446-radio/SoftwareModem.ipynb).
I tried Rattlegram with two HTs and that worked as well, so I'll keep on experimenting.

Rattlegram sends shorter messages than the 5380bytes of this modem.  Is there any way to send shorter messages with this modem as well?  I'd like to reduce air time.  In most cases messages will be much shorter than 5380bytes.
